### PR TITLE
Share popup with permalink and citation

### PR DESCRIPTION
**Pull request by jrchudy**
This PR adds support for the citation feature in `record` app.

The permalink button has been removed with the permalink being a part of the new "share" popup. A share button has been added to the bookmark bar in it's place. After clicking the share button, a modal dialog should appear with "Share Citation" as it's title.

There should be 2 elements in an unordered list (if a citation annotation exists for the table, 1 if not). The first should say "Share Link" with the permalink listed below it. The second list item should say "Citation" with the citation listed below it.

[This link](https://dev.isrd.isi.edu/chaise/record/#49462/faceting:main/RID=1SC) (schema:table) has the citation annotation defined. @RFSH, my user directory is going to change as I test other PRs and start working on the next task. But you should be able to look at the table still after updating the code in your user directory. The catalog will expire (and be deleted) on Thursday.

### CODE REVIEW

**Comment by jrchudy** [comment]
This PR relies on the corresponding changes in `ermrestJS`. That PR must be merged first before the tests in this branch will pass.

File: record/index.html.in [hunk 1]
```diff
@@ -39,8 +39,8 @@
                         Related Records
                     </strong>
                 </a>&nbsp;&nbsp;
-                <a id="permalink" ng-href="{{::ctrl.permalink()}}" uib-tooltip={{::ctrl.tooltip.permalink}} tooltip-placement="bottom">
-                    <span class="glyphicon glyphicon-bookmark"></span><strong> Permalink</strong>
+                <a id="share" ng-click="::ctrl.sharePopup()" uib-tooltip="Some text" tooltip-placement="bottom">
```
Comment (RFSH) [inline record/index.html.in]: I think you forgot to update the tooltip.

File: test/e2e/specs/all-features-confirmation/record/presentation.spec.js [hunk 2]
```diff
@@ -164,6 +169,7 @@ describe('View existing record,', function() {
             keys.push(testParams.key.name + testParams.key.operator + testParams.key.value);
             browser.ignoreSynchronization=true;
             var url = browser.params.url + "/record/#" + browser.params.catalogId + "/product-record:" + testParams.table_name + "/" + keys.join("&");
+            testParams.citationParams.permalink = url;
```
Comment (RFSH) [inline test/e2e/specs/all-features-confirmation/record/presentation.spec.js]: Is there a better way that we could test this? The definition of `permalink` is separated from other attributes and it's going to be difficult to follow the test case. Why not just test it with the current browser location?

Comment (jrchudy) [reply test/e2e/specs/all-features-confirmation/record/presentation.spec.js]: I didn't even consider that case. That's the right thing to test. I will update the other test case with the same.

**Comment by jrchudy** [comment]
@RFSH I made the changes you suggested. Should merge this PR?

**Pull request merged**